Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docrepository/adapter/dao/DocumentDao.java

```diff
@@ -102,176 +104,179 @@ protected Session getSession() {
      * @param params Query parameters
      * @return Query results
      */
-    @SuppressWarnings("unchecked")
-    public List<Document> findDocuments(DocumentQueryParams params) {
+    public List<DocumentMetadata> findDocuments(DocumentQueryParams params) {
         LOG.debug("Beginning document query");
 
-        String patientId = null;
-        List<String> classCodes = null;
-        String classCodeScheme = null;
-        Date creationTimeFrom = null;
-        Date creationTimeTo = null;
-        Date serviceStartTimeFrom = null;
-        Date serviceStartTimeTo = null;
-        Date serviceStopTimeFrom = null;
-        Date serviceStopTimeTo = null;
-        List<String> statuses = null;
-        List<String> documentUniqueIds = null;
-        if (params != null) {
-            patientId = params.getPatientId();
-            classCodes = params.getClassCodes();
-            classCodeScheme = params.getClassCodeScheme();
-            creationTimeFrom = params.getCreationTimeFrom();
-            creationTimeTo = params.getCreationTimeTo();
-            serviceStartTimeFrom = params.getServiceStartTimeFrom();
-            serviceStartTimeTo = params.getServiceStartTimeTo();
-            serviceStopTimeFrom = params.getServiceStopTimeFrom();
-            serviceStopTimeTo = params.getServiceStopTimeTo();
-            statuses = params.getStatuses();
-            documentUniqueIds = params.getDocumentUniqueIds();
+        DocumentQueryParams parameters = params;
```

Comment:
Do we need to create parameter here?  Recommend to remove it

Reply:
Not sure about you, but I find that one line who's getters will be defaulted to null is much easier to read than the 11 different variables above it being set. Later on we do null checks when adding the criteria anyway so its silly not do just give it a default new object.